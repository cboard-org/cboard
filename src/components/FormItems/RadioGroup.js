import React from 'react';
import { RadioGroup as MUIRadioGroup } from 'material-ui/Radio';
import { FormControl, FormLabel } from 'material-ui/Form';

const RadioGroup = ({ children, className, label, error, ...rest }) => (
  <FormControl className={className} error={!!error}>
    <FormLabel>{label}</FormLabel>
    <MUIRadioGroup {...rest}>{children}</MUIRadioGroup>
  </FormControl>
);

export default RadioGroup;
