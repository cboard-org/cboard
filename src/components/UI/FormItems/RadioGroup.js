import React from 'react';
import PropTypes from 'prop-types';
import { RadioGroup as MUIRadioGroup } from '@material-ui/core/Radio';
import FormControl from '@material-ui/core/FormControl';
import FormLabel from '@material-ui/core/FormLabel';

const propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  className: PropTypes.string,
  error: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number
  ]),
  label: PropTypes.string
};

const defaultProps = {
  className: '',
  error: false,
  label: ''
};

const RadioGroup = ({ children, className, error, label, ...rest }) => (
  <FormControl className={className} error={!!error}>
    <FormLabel>{label}</FormLabel>
    <MUIRadioGroup {...rest}>{children}</MUIRadioGroup>
  </FormControl>
);

RadioGroup.propTypes = propTypes;
RadioGroup.defaultProps = defaultProps;

export default RadioGroup;
