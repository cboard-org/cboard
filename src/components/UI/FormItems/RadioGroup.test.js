import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import RadioGroup from './RadioGroup';

describe('RadioGroup tests', () => {
  test('default renderer', () => {
    const children = [
      <div key="1">"first"</div>,
      <div key="2">"second"</div>,
      <div key="3">"third"</div>
    ];
    shallowMatchSnapshot(<RadioGroup>{children}</RadioGroup>);
  });
});
