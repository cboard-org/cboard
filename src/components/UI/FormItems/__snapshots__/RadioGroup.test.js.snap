// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup tests default renderer 1`] = `
<WithStyles(FormControl)
  className=""
  error={false}
>
  <WithStyles(FormLabel)>
    
  </WithStyles(FormLabel)>
  <UNDEFINED>
    <div>
      "first"
    </div>
    <div>
      "second"
    </div>
    <div>
      "third"
    </div>
  </UNDEFINED>
</WithStyles(FormControl)>
`;
